SalesforceSDK/Salesforce.SDK.App/App/SalesforceApplication.cs

```diff
@@ -139,7 +139,9 @@ private void CoreVisibilityChanged(CoreWindow sender, VisibilityChangedEventArgs
         {
             try
             {
-                await OAuth2.RefreshAuthToken(AccountManager.GetAccount());
+                //Assign this to a var as ref requires it.
+                var account = AccountManager.GetAccount();
+                await OAuth2.RefresAuthToken(account);
                 OAuth2.RefreshCookies();
             }
             catch (WebException ex)
```

Comment:
You're changing the exception to OauthException at this point, need to catch that instead of WebException.